app/dashboard/page.tsx

```diff
@@ -0,0 +1,291 @@
+              });
+              if (hexToDecimal(address) === hexToDecimal(data.addr))
+                setIsOwner(true);
+              setInitProfile(true);
+            });
+          })
+          .catch(() => {
+            return;
+          });
+      } else {
+        starknetIdNavigator
+          ?.getAddressFromStarkName(addressOrDomain)
+          .then((addr) => {
+            setIdentity({
+              starknet_id: "0",
+              addr: addr,
+              domain: addressOrDomain,
+              is_owner_main: false,
+            });
+            setInitProfile(true);
+            if (hexToDecimal(address) === hexToDecimal(addr)) setIsOwner(true);
+          })
+          .catch(() => {
+            return;
+          });
+      }
+    } else if (
+      typeof addressOrDomain === "string" &&
+      isHexString(addressOrDomain)
+    ) {
+      starknetIdNavigator
+        ?.getStarkName(hexToDecimal(addressOrDomain))
+        .then((name) => {
+          if (name) {
+            if (
+              !utils.isBraavosSubdomain(name) &&
+              !utils.isXplorerSubdomain(name)
+            ) {
+              starknetIdNavigator
+                ?.getStarknetId(name)
+                .then((id) => {
+                  getIdentityData(id).then((data: Identity) => {
+                    if (data.error) return;
+                    setIdentity({
+                      ...data,
+                      starknet_id: id.toString(),
+                    });
+                    if (hexToDecimal(address) === hexToDecimal(data.addr))
+                      setIsOwner(true);
+                    setInitProfile(true);
+                  });
+                })
+                .catch(() => {
+                  return;
+                });
+            } else {
+              setIdentity({
+                starknet_id: "0",
+                addr: addressOrDomain,
+                domain: name,
+                is_owner_main: false,
+              });
+              setInitProfile(true);
+              if (hexToDecimal(addressOrDomain) === hexToDecimal(address))
+                setIsOwner(true);
+            }
+          } else {
+            setIdentity({
+              starknet_id: "0",
+              addr: addressOrDomain,
+              domain: minifyAddress(addressOrDomain),
+              is_owner_main: false,
+            });
+            setIsOwner(false);
+            setInitProfile(true);
+          }
+        })
+        .catch(() => {
+          setIdentity({
+            starknet_id: "0",
+            addr: addressOrDomain,
+            domain: minifyAddress(addressOrDomain),
+            is_owner_main: false,
+          });
+          setInitProfile(true);
+          if (hexToDecimal(addressOrDomain) === hexToDecimal(address))
+            setIsOwner(true);
+        });
+    } else {
+      setNotFound(true);
+    }
+  }, [addressOrDomain, address]);
+
+  const getIdentityData = async (id: string) => { 
```

Comment:
can you add this in a seperate file called starknedId.service inside the `services` folder. we couple all api services in one file